### PR TITLE
Fix occasional freeze bug

### PR DESCRIPTION
**Pull request by LogicParrot**
This might close the player freeze issue (#3130). I could not reproduce the problem locally, so I did not test this. Can someone test this and report back? @popsUlfr @Pokechu22

Binaries are available below. You can also fetch the `freezeFix` branch and compile this.
- [Linux x64](https://builds.cuberite.org/job/Cuberite%20Linux%20x64%20ingame-testing-needed/11/)
- [Raspberry Pi](https://builds.cuberite.org/job/Cuberite%20Linux%20raspi-armhf%20ingame-testing-needed/10/)
- [Windows x64](https://builds.cuberite.org/job/Cuberite%20Windows%20x64%20ingame-testing-needed/10/)

### CODE REVIEW

**Comment by LogicParrot** [comment]
I managed to test this. It works.

File: src/Entities/Player.cpp [hunk 1]
```diff
@@ -65,6 +65,7 @@ cPlayer::cPlayer(cClientHandlePtr a_Client, const AString & a_PlayerName) :
 	m_GameMode(eGameMode_NotSet),
 	m_IP(""),
 	m_ClientHandle(a_Client),
+	m_IsFrozen(false),
```
Comment (LogicParrot) [inline src/Entities/Player.cpp]: This is the only change. The rest is minor code tidying.

**Comment by popsUlfr** [comment]
Thanks @LogicParrot, I can't reproduce the issue on my end anymore ;)

File: src/Entities/Player.cpp (hunk 1 above)
Comment (tigerw) [inline src/Entities/Player.cpp]: m_IsManuallyFrozen?

Comment (LogicParrot) [inline src/Entities/Player.cpp]: The value of `m_IsManuallyFrozen` is irrelevant when `m_IsFrozen` is `false`.

**Comment by LogicParrot** [comment]
Glad it works :)

For the record: `m_IsManuallyFrozen` and `m_IsFrozen` were both unitialized prior to this. So they are randomly true or false initially. The bug triggers when both `m_IsFrozen` and `m_IsManuallyFrozen` are initialized to `true`. Assuming equal chance for true and false, this bug would have manifested itself in 25% of all logins.

File: src/Entities/Player.cpp (hunk 1 above)
Comment (LogicParrot) [inline src/Entities/Player.cpp]: Hmm. Perhaps it would be wiser to have an enum. `m_FrozenState { None, Manual, Auto }`

Comment (tigerw) [inline src/Entities/Player.cpp]: An excellent idea.

**Pull request merged**